SolrPhpClient/Apache/Solr/Service.php

```diff
@@ -330,8 +330,11 @@ protected function _sendRawGet($url, $timeout = FALSE)
 		}
 		
 		// Islandora: dump solr query address in debug mode
-		if ( variable_get('islandora_solr_debug_mode', 0) ) drupal_set_message(l('solr query',$url."&indent=on&debugQuery=true"));
-
+		if (variable_get('islandora_solr_debug_mode', 0)) {
+                  if (user_access('view solr debug')) {
```

Comment:
This can be done in one conditional:

`if (variable_get('islandora_solr_debug_mode', 0) && user_access('view solr debug'))`